tests/src/Interop/IJW/NativeCallingManaged/IjwNativeCallingManagedDll/CMakeLists.txt

```diff
@@ -0,0 +1,46 @@
+cmake_minimum_required (VERSION 2.6)
+project (IjwNativeCallingManagedDll)
+include_directories( ${INC_PLATFORM_DIR} )
+set(SOURCES IjwNativeCallingManagedDll.cpp)
+
+if (WIN32)
+  # 4365 - signed/unsigned mismatch
+  add_compile_options(-wd4365)
+
+  # IJW
+  add_compile_options(-clr)
+  
+  # IJW requires the CRT as a dll, not linked in
+  if(UPPERCASE_CMAKE_BUILD_TYPE STREQUAL DEBUG OR UPPERCASE_CMAKE_BUILD_TYPE STREQUAL CHECKED)
```

Comment:
This will not work as expected. For Windows, CMake uses so called multi-config generator to generate the VS project files. That means that the project files support all configurations at once. The CMake scripts are processed just once and so we won't get here once with build type set to DEBUG, once with it set to RELEASE and then with checked.
For multi-config generators, you need to use so called generator expression here::
```cmake
add_compile_options(-MD$<$<OR:$<CONFIG:Release>,$<CONFIG:Checked>>:d>)
```
An additional nit - to stay in line with the other options specified, use the '/' instead of '-'?

Reply:
Fixed